resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala

```diff
@@ -127,6 +121,11 @@ private[spark] class Client(
             .pods()
             .withLabels(driverKubernetesSelectors)
```

Comment:
We would need an extra layer of nesting to do that, since we want to delay the service creation for as long as possible. So we want to only make the service after we make the pod, so we'll need a `Watch` for the pod being created to trigger creating the service, and another watch for the `Service` having the backing pod. It's doable, just not sure how to write it in a way that makes the nesting easy to reason about.